### PR TITLE
Clone Google Test and RapidJSON if Not Found

### PR DESCRIPTION
**Pull request by iguessthislldo**

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -1365,7 +1365,25 @@ if (exists $opts{'gtest'} || $tests) {
...
                               $is_windows ? 'googletest-distribution' : ''));
 
+    my $use_sm_dir = 0;
     if (-f File::Spec->catfile($sm_src, $gtest_sanity)) {
+      $use_sm_dir = 1;
+    }
+    elsif (-f File::Spec->catfile($sys_dir, $gtest_sanity)) {
+      $opts{'gtest'} = $sys_dir;
+      $gtest_root = $sys_dir;
+      setEnv('GTEST_ROOT', $gtest_root);
```
Comment (simpsont-oci) [inline configure:1375]: Does this setEnv() need a print statement as well (similar to below?)

**Review by simpsont-oci — APPROVED** [review]

File: configure (hunk 1 above)
Comment (iguessthislldo) [reply configure:1375]: It looks like it didn't because it's a special case for safety profile.

File: configure [hunk 2]
```diff
@@ -425,6 +425,38 @@ sub run_command {
...
+  my $branch = shift;
+  print "Cloning git repo $url branch $branch\n";
+  my $rv = run_command(['git', 'clone', '-q', '--depth=1', '-b', $branch, $url, $dest]);
+  if (!$rv) {
+    run_command(['git', '--no-pager', 'log', '-1', '--oneline'], chdir => $dest);
+  }
+  return $rv;
+}
+
+sub git_sm_prop {
```
Comment (mitza-oci) [inline configure:440]:
```suggestion
sub git_submodule_prop {
```

Or change the error message on 445, but I think the longer function name is more clear

**Review by mitza-oci — CHANGES_REQUESTED** [review]
This part of the configure script could be tested in GitHub Actions by having at least one job that doesn't use built-in git submodule support (or just deletes the one that are already there).

**Review by mitza-oci — APPROVED** [review]

**Pull request merged**